src/components/Modal.tsx

```diff
@@ -0,0 +1,48 @@
+import { Dialog, Transition } from '@headlessui/react';
+import React from 'react';
+export type ModalProps = {
+  children: React.ReactNode;
+  isOpen: boolean;
+  onClose: () => void;
+  bg?: string;
+};
+export default function Modal({
+  children,
+  isOpen,
+  onClose,
+  bg = 'bg-black/25',
```

Comment:
nice, thanks for refactoring this!

Reply:
You might be interested in this: https://catalyst.tailwindui.com/

Reply:
Oh wow, it looks really nice...will definitely consider it moving forward! (not too willing to use it rn since its still in dev 😅 )